drakrun/drakrun/test/test_network.py

```diff
@@ -4,45 +4,116 @@
     delete_vm_network,
     iptable_rule_exists,
     start_dnsmasq,
-    stop_dnsmasq
+    stop_dnsmasq,
+    add_iptable_rule,
+    del_iptable_rule,
+    start_tcpdump_collector
 )
 
 from drakrun.draksetup import find_default_interface
 import os
 import subprocess
+from pathlib import Path
+from drakrun.test.common_utils import tool_exists
 
 
-def tool_exists(tool):
-    return subprocess.run(["which", tool]).returncode == 0
+def count_num_rules(rule_to_check):
+    rules = subprocess.run(['iptables', '-S'], capture_output=True).stdout.decode().split('\n')
+
+    # Arguments used in iptables -A are being split
+    # as the arguments order doesn't remain the same in iptables -S and iptables -A
+    arguments_to_search = rule_to_check.split('-')
+    count = 0
+    for rule in rules:
+
+        # Assume all the terms are present
+        flag = True
+
+        for argument in arguments_to_search:
+            if argument not in rule:
+                # if one of the argument is not found, the rule must be different
+                flag = False
+
+        # if all arguments there, add it in similar rule
+        if flag is True:
+            count += 1
+
+    return count
+
+
+@pytest.mark.skipif(not tool_exists('iptables'), reason="iptables does not exist")
+def test_iptables():
+    rule = "INPUT -i draktest0 -d 239.255.255.0/24 -j DROP"
+
+    assert iptable_rule_exists(rule) is False
+
+    add_iptable_rule(rule)
+    assert iptable_rule_exists(rule) is True
+
+    # adding second time also
+    add_iptable_rule(rule)
+
+    # it should not be added second time
+    assert count_num_rules(rule) == 1
+
+    # if somehow added due to unknown issues
+
+    # this call is adding the rule again to test if del_iptable_rule does delete multiple similar rules or not
+    subprocess.check_output(f"iptables -A {rule}", shell=True)
+
+    # the clear should delete all the same rules
+    del_iptable_rule(rule)
+    assert iptable_rule_exists(rule) is False
```

Comment:
please use `assert not` instead of `assert .. is False`

Reply:
Please add some lint checks for enforcing such code styles if possible, there aren't really intuitive for me